dev/tools/bin/generate_gradle_lockfiles.dart

```diff
@@ -0,0 +1,183 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This script generates `android/build.gradle` for each directory specify in the stdin.
+// Then it generate the lockfiles for each Gradle project.
+// To regenerate these files, run `find . -type d -name 'android' | dart dev/tools/bin/generate_gradle_lockfiles.dart`
+
+import 'dart:io';
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:path/path.dart' as path;
+
+void main(List<String> arguments) {
+  print(
+      "Usage: find . -type d -name 'android' | dart dev/tools/bin/generate_gradle_lockfiles.dart\n\n");
```

Comment:
nit: no newline.

Reply:
also nit: this makes it harder for developers on windows to run this script. Consider changing this to a file path argument where we just read the file.

Reply:
what would be the issue?  I have added a section about how to enter the files manually to write to the stdin pipe.

FWIW,  the blaze command (internally) works this way for running multiple targets at once. I don't know what part wouldn't work on Windows since powershell also has support for the pipe operator.

Reply:
Ahh ok, so I guess on Windows you could just use something like `gci -Recurse -Directory -Name "android" | dart ...`. That should be fine then.

Reply:
🎉